dali/base/dafdesc.cpp

```diff
@@ -1353,14 +1358,26 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
                 *_interleaved = interleaved;
         }
 
-        const char * historyStr = "History";
-        if (mb.remaining() > strlen(historyStr))
+        // If we have more serialised data in MemoryBuffer
+        if (mb.remaining() > sizeof(extendedVersion))
         {
-            const char * buffer = mb.toByteArray()+mb.getPos();
-            if (nullptr != strstr(buffer, historyStr))
-                history.setown(createPTree(mb));
+            mb.read(extendedVersion);
+
+            if (extendedVersion)
+            {
+                //If extendedVersion >= 1 then we should have stored History
+                const char * historyStr = "History";
+                if (mb.remaining() > strlen(historyStr))
```

Comment:
It's now an error if it is not what's expected, so should fire exceptions just as you do if version=0 if anything doesn't match. In fact you should assume it does, just as everything else does.
You've read version it's >= MIN , now deserialize ptree without further validation.

And to make that crystal clear, the block could start with:
```
if (extendedVersion >= VERSION_MINHISTORY)
```